receiver/mongodbatlasreceiver/internal/mongodb_atlas_client.go

```diff
@@ -17,14 +17,66 @@ package internal // import "github.com/open-telemetry/opentelemetry-collector-co
 import (
 	"context"
 	"fmt"
+	"net/http"
+	"time"
 
+	"github.com/cenkalti/backoff/v4"
 	"github.com/mongodb-forks/digest"
 	"github.com/pkg/errors"
 	"go.mongodb.org/atlas/mongodbatlas"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
 	"go.opentelemetry.io/collector/model/pdata"
 	"go.uber.org/zap"
 )
 
+type RetryBackoffRoundTripper struct {
+	originalTransport http.RoundTripper
+	log               *zap.Logger
+	retrySettings     exporterhelper.RetrySettings
+}
+
+func (rt *RetryBackoffRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+
+	resp, err := rt.originalTransport.RoundTrip(r)
+	if err != nil {
+		return nil, err // Can't do anything
+	}
+
+	if resp.StatusCode == 429 {
+		expBackoff := &backoff.ExponentialBackOff{
+			InitialInterval:     rt.retrySettings.InitialInterval,
+			RandomizationFactor: backoff.DefaultRandomizationFactor,
+			Multiplier:          backoff.DefaultMultiplier,
+			MaxInterval:         rt.retrySettings.MaxInterval,
+			MaxElapsedTime:      rt.retrySettings.MaxElapsedTime,
+			Stop:                backoff.Stop,
+			Clock:               backoff.SystemClock,
+		}
+		expBackoff.Reset()
+		attempts := 0
+		for {
+			attempts++
+			delay := expBackoff.NextBackOff()
+			if delay == backoff.Stop {
+				return resp, err
+			}
+			rt.log.Warn("server busy, retrying request",
+				zap.Int("attempts", attempts),
+				zap.Duration("delay", delay))
+			time.Sleep(delay)
```

Comment:
This will block the Collector when shutting down. Maybe we want an approach similar to https://github.com/open-telemetry/opentelemetry-collector/blob/18967454aa63c8a21ec50750daa532e2d8e633e0/exporter/exporterhelper/queued_retry.go#L222-L229 ?